go/httpdispatcher/proxy.go

```diff
@@ -0,0 +1,312 @@
+				<Redirect>
+					<MessageType>100</MessageType>
+					<ResponseCode>0</ResponseCode>
+					<AccessProcedure>1.0</AccessProcedure>
+					<VersionLow>1.0</VersionLow>
+					<VersionHigh>2.0</VersionHigh>
+					<AccessLocation>CDATA[[isocc=,cc=,ac=,network=PacketFence,]]</AccessLocation>
+					<LocationName>CDATA[[PacketFence]]</LocationName>
+					<LoginURL>{{.WisprUrl.String}}</LoginURL>
+				</Redirect>
+			</WISPAccessGatewayParam>-->
+	</body>
+</html>`)
+		t.Execute(w, passThrough)
+
+		//spew.Dump(r)
+		//spew.Dump(r.Header.Get("X-Forwarded-Proto"))
+		log.LoggerWContext(ctx).Debug(fmt.Sprintln(host, "REDIRECT"))
+		return
+	}
+	if !p.checkEndpointList(ctx, host) {
+		log.LoggerWContext(ctx).Info(fmt.Sprintln(host, "FORBIDDEN"))
+		w.WriteHeader(http.StatusForbidden)
+		return
+	}
+	log.LoggerWContext(ctx).Debug(fmt.Sprintln(host, "REVERSE"))
+	rp := httputil.NewSingleHostReverseProxy(&url.URL{
+		Scheme: "http",
+		Host:   host,
+	})
+	t := time.Now()
+
+	// Pass the context in the request
+	r = r.WithContext(ctx)
+	rp.ServeHTTP(w, r)
+	log.LoggerWContext(ctx).Info(fmt.Sprintln(host, time.Since(t)))
+}
+
+// Configure add default target in teh deny list
+func (p *Proxy) Configure(ctx context.Context, port string) {
+	p.addToEndpointList(ctx, "localhost")
+	p.addToEndpointList(ctx, "127.0.0.1")
+}
+
+func (p *passthrough) readConfig(ctx context.Context) {
+	var trapping pfconfigdriver.PfConfTrapping
+	var portal pfconfigdriver.PfConfCaptivePortal
+	var general pfconfigdriver.PfConfGeneral
+	var scheme string
+
+	pfconfigdriver.FetchDecodeSocketStruct(ctx, &trapping)
+	pfconfigdriver.FetchDecodeSocketStruct(ctx, &portal)
+	pfconfigdriver.FetchDecodeSocketStruct(ctx, &general)
+
+	for _, v := range trapping.ProxyPassthroughs {
+		p.addFqdnToList(ctx, v)
+	}
+
+	for _, v := range portal.DetectionMecanismUrls {
+		p.addDetectionMechanismsToList(ctx, v)
+	}
+
+	//p.Cache = cache.New(3*time.Second, 1*time.Second)
+
+	p.DetectionMecanismBypass = portal.DetectionMecanismBypass
+	if portal.DetectionMecanismBypass == "enabled" {
+		p.DetectionMecanismBypass = true
+	}
+	if portal.DetectionMecanismBypass == "disabled" {
+		p.DetectionMecanismBypass = false
+	}
+	rgx, _ := regexp.Compile("CaptiveNetworkSupport")
+	p.URIException = rgx
+	if portal.SecureRedirect == "enabled" {
+		p.SecureRedirect = true
+		scheme = "https"
+	}
+	if portal.SecureRedirect == "disabled" {
+		p.SecureRedirect = false
+		scheme = "http"
+	}
+	var portalURL url.URL
+	var wisprURL url.URL
+
+	portalURL.Host = general.Hostname + "." + general.Domain
+	portalURL.Path = "/captive-portal"
+	portalURL.Scheme = scheme
+
+	wisprURL.Host = general.Hostname + "." + general.Domain
+	wisprURL.Path = "/wispr"
+	wisprURL.Scheme = scheme
+
+	p.WisprURL = &wisprURL
+	//general.Hostname + "." + general.Domain + "/wispr"
+	p.PortalURL = &portalURL
+	//general.Hostname + "." + general.Domain + "/captive-portal"
+
+}
+
+func newProxyPassthrough(ctx context.Context) *passthrough {
```

Comment:
passthrough struct and functions aren't documented. Just a few lines should be enough